src/coreclr/src/System.Private.CoreLib/src/Internal/Runtime/InteropServices/ComponentActivator.cs

```diff
@@ -101,21 +98,38 @@ private static string MarshalToString(IntPtr arg, string argName)
             return 0;
         }
 
-        private static Delegate CreateDelegate(string assemblyPath, string typeName, string methodName, string delegateTypeName)
+        private static IntPtr GetFunctionPointer(string assemblyPath, string typeName, string methodName, string? delegateTypeName)
         {
-            // Throws
             IsolatedComponentLoadContext alc = GetIsolatedComponentLoadContext(assemblyPath);
-
             Func<AssemblyName, Assembly> resolver = name => alc.LoadFromAssemblyName(name);
 
-            // Throws
             Type type = Type.GetType(typeName, resolver, null, throwOnError: true)!;
 
-            // Throws
-            Type delegateType = Type.GetType(delegateTypeName, resolver, null, throwOnError: true)!;
+            IntPtr functionPtr;
+            if (delegateTypeName == null)
+            {
+                MethodInfo? methodInfo = type.GetMethod(methodName);
+                if (methodInfo == null)
+                    throw new MissingMethodException(typeName, methodName);
+
+                Debug.Assert(null != methodInfo.GetCustomAttribute<UnmanagedCallersOnlyAttribute>());
+                functionPtr = ((RuntimeMethodInfo)methodInfo).MethodHandle.GetFunctionPointer();
```

Comment:
```suggestion
                functionPtr = methodInfo.MethodHandle.GetFunctionPointer();
```

`MethodHandle` is a method MethodInfo. The cast should not be needed.